packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -561,3 +568,38 @@ export class PerformanceEvent {
         throw error;
     }
 }
+
+/**
+     * Take in a event object, stringify any fields that are non-primitives, and return the new event object.
+     * @param event - Event with fields you want to stringify.
+     */
+function stringifyEventFields(event: TelemetryEventTypes): void {
```

Comment:
The purpose here is to conform the event to `ITelemetryBaseEvent`, right?  And furthermore, in `sendTelemetryEventCore` the first thing it does is clone the event via `{ ...event }`.  So you can take that opportunity to transform the event as it's cloned, but updating this function to return a new object with the correct type, something like this:

`function convertToBaseEvent(event: TelemetryEventTypes): ITelemetryBaseEvent { ... }`

And it will look really similar to this but rather than overwriting `event[key]` it will copy over to the returned event.